modules/javafx.graphics/src/main/java/javafx/scene/SpotLight.java

```diff
@@ -0,0 +1,239 @@
+ * </ul>
+ * The valid ranges for these properties are {@code 0 <= innerAngle <= outerAngle <= 180} and {@code falloff >= 0};
+ * values outside either of these ranges can produce unexpected results.
+ * <p>
+ * For a point whose vector to the light forms an angle {@code theta} with the direction of the light, if
+ * <ul>
+ * <li>{@code theta < innerAngle} then that point receives maximum illumination
+ * <li>{@code theta > outerAngle} then that point receives no illumination
+ * <li>{@code innerAngle <= theta <= outerAngle} then the illumination is determined by the formula
+ *
+ * <pre>I = pow((cos(theta) - cos(outer)) / (cos(inner) - cos(outer)), falloff)</pre>
+ *
+ * which represents a drop in illumination from the inner angle to the outer angle.
+ * </ul>
+ * <p>
+ * <img src="doc-files/spotlight.png" alt="Image of the Spotlight">
+ *
+ * @since 17
+ * @see PhongMaterial
+ */
+public class SpotLight extends PointLight {
+    static {
+        SpotLightHelper.setSpotLightAccessor(new SpotLightHelper.SpotLightAccessor() {
+            @Override
+            public NGNode doCreatePeer(Node node) {
+                return ((SpotLight) node).doCreatePeer();
+            }
+
+            @Override
+            public void doUpdatePeer(Node node) {
+                ((SpotLight) node).doUpdatePeer();
+            }
+        });
+    }
+
+    {
+        // To initialize the class helper at the beginning of each constructor of this class
+        SpotLightHelper.initHelper(this);
+    }
+
+    /**
+     * Creates a new instance of {@code SpotLight} class with a default {@code Color.WHITE} light source.
+     */
+    public SpotLight() {
+        super();
+    }
+
+    /**
+     * Creates a new instance of {@code SpotLight} class using the specified color.
+     *
+     * @param color the color of the light source
+     */
+    public SpotLight(Color color) {
+        super(color);
+    }
+
+
+    /**
+     * The direction vector of the spotlight. The vector need not be normalized.
+     *
+     * @defaultValue {@code Point3D(0, 0, 1)}
+     */
+    private ObjectProperty<Point3D> direction;
+
+    public final void setDirection(Point3D value) {
+        directionProperty().set(value);
+    }
+
+    private static final Point3D DEFAULT_DIRECTION = NGSpotLight.getDefaultDirection();
+
+    public final Point3D getDirection() {
+        return direction == null ? DEFAULT_DIRECTION : direction.get();
+    }
+
+    public final ObjectProperty<Point3D> directionProperty() {
+        if (direction == null) {
+            direction = new SimpleObjectProperty<>(this, "direction", DEFAULT_DIRECTION) {
+                @Override
+                protected void invalidated() {
+                    NodeHelper.markDirty(SpotLight.this, DirtyBits.NODE_LIGHT);
+                }
+            };
+        }
+        return direction;
+    }
+
+
+    /**
+     * The angle of the spotlight's inner cone. Surfaces whose angle to the light's origin is less than this angle
```

Comment:
angle of a cone is angle between the two sides of cone or the total angle of cone. Here we are referring to an angle between the Light's direction vector and any side of cone. which would be half of the angle of a cone.
So should we mention it here ? 
1. Half of the inner cone angle Or
2. Angle between the direction vector of light and edge/side of cone.

Am I over thinking here ?

Or we should not describe the behavior in terms of angle but mention it in terms of location/distance of a point. If point is inside the cone or outside the cone, distance of the point from the centre of cone.

Reply:
I think that it's simply best to refer the reader to the image in the class doc.